### PR TITLE
Old channle9 video link removed

### PR DESCRIPTION
**Pull request by v-hearya**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/visual-basic/getting-started/additional-resources.md
Comment (IEvangelist) [inline docs/visual-basic/getting-started/additional-resources.md:42]:
```suggestion
|[GitHub: Visual Basic .NET](https://github.com/dotnet/vblang)|The official repo for Visual Basic .NET language design.|
```

File: docs/visual-basic/programming-guide/concepts/async/index.md
Comment (IEvangelist) [inline docs/visual-basic/programming-guide/concepts/async/index.md:245]:
```suggestion
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
This PR needs a bit more love, while it does remove a lot of old Channel 9 links there is additional text that would also need to be removed on a case-by-case basis. I've made suggestions to help improve this, thank you

File: docs/csharp/async.md
Comment (BillWagner) [reply docs/csharp/async.md:266]: This one will fix #27287

File: docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv
Comment (BillWagner) [inline docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv:600]: Ping @luisquintanilla 

I think this should be reverted. This file is used for training an ML.NET sample. It's not used in docs.

However, I defer to @luisquintanilla for his opinion.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks @v-hearya 

I have some suggestions as well before this is merged.

Also, adding @luisquintanilla as a reviewer.

File: docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv:600]: Thanks @BillWagner. That is correct. This is not documentation but a data file used to train an ML model as part of a tutorial.  Please revert.

**Comment by ShannonLeavitt** [comment]
I'm starting revisions to this content now. I'll commit suggestions and resolve merge conflicts.

File: docs/architecture/microservices/docker-application-development-process/docker-app-development-workflow.md
Comment (ShannonLeavitt) [inline docs/architecture/microservices/docker-application-development-process/docker-app-development-workflow.md:547]:
```suggestion

```
Removing empty section

File: docs/csharp/async.md
Comment (ShannonLeavitt) [inline docs/csharp/async.md:262]:
```suggestion
* [The Task asynchronous programming model (C#)](./programming-guide/concepts/async/task-asynchronous-programming-model.md).

```

File: docs/framework/mef/attributed-programming-model-overview-mef.md
Comment (ShannonLeavitt) [inline docs/framework/mef/attributed-programming-model-overview-mef.md:977]:
```suggestion
`IPartImportsSatisfiedNotification` contains one method named `OnImportsSatisfied`. This method is called by the composition container on any parts that implement the interface when composition has been completed and the part's imports are ready for use. Parts are created by the composition engine to fill the imports of other parts. Before the imports of a part have been set, you cannot perform any initialization that relies on or manipulates imported values in the part constructor unless those values have been specified as prerequisites by using the `ImportingConstructor` attribute. This is normally the preferred method, but in some cases, constructor injection may not be available. In those cases, initialization can be performed in `OnImportsSatisfied`, and the part should implement `IPartImportsSatisfiedNotification`.

```

File: docs/standard/analyzers/portability-analyzer.md
Comment (ShannonLeavitt) [inline docs/standard/analyzers/portability-analyzer.md:110]:
```suggestion
For more information on the .NET Portability Analyzer, visit the [GitHub documentation](https://github.com/Microsoft/dotnet-apiport#documentation).

```

File: docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv
Comment (ShannonLeavitt) [inline docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv:600]:
```suggestion
2329 area-System.Threading Use a more dependable policy for thread pool thread injection As of .NET 4.5 the thread pool injects one thread per 0.5sec if it thinks more threads are required. This is a problem if the number of required threads suddenly increases. Example: A corporate ASP.NET website is idle at night. In the morning at 8AM 1000 people log on and start working. If the app is using let's say 100 threads starting from 8AM it will take like 50sec to create all of them. Until then there will be serious delays and timeouts. It is possible to construe arbitrarily bad scenarios. **Problem statement:** If thread pool load suddenly increases in IO bound workloads the pool is too slow to respond. This causes throughput and availability disruption. IO bound workloads relying on synchronous IO are common. Sudden workload changes are also common. Sometimes the workload can change due to a problem outside of the developer's control: A web service timing out or a database become slow. Let me stress that this causes service interruption. You can easily repro this yourself. Run a load test on an ASP.NET site with `Thread.Sleep(10000);`. The thread count goes up by 2 each second. Starting and shutting down a thread was benchmarked by me to be around 1ms in total. Threads are not really an expensive resource. The thread pool should be a lot more eager to create and destroy threads. 500ms delay to potentially save 1ms is not a good trade-off. **Easy fix:** I propose lowering the injection delay to 100ms. This reduces the problem given above by 5x. Ideally, the rate would be configurable. The shutdown delay could be lowered from 30s as well. Keeping an idle thread for 30000ms to save 1ms seems excessive. In this ticket I'm not that concerned with retiring threads, though. **Smarter, riskier fix:** The delay could depend on the number of threads in existence, the core count and the perceived pressure on the thread-pool. The injection rate could be: - 0ms delay for up to (ProcessorCount \* 1) threads - 50ms delay for up to (ProcessorCount \* 4) threads - Starting from that a delay of (100ms \* (ThreadCount / ProcessorCount \* someFloatFactor)). Reasoning: The more the CPU is oversubscribed the slower we want to inject. Maybe we need to have a maximum delay of 1sec. Or, the delay must rise sub-linearly (e.g. sqrt). Note, that the Windows Kernel has some thread pool injection heuristics that apply back pressure the more threads are in existence. This seems to work. Source: https://channel9.msdn.com/Shows/Going+Deep/Inside-Windows-8-Pedro-Teixeira-Thread-pool 

2332 area-System.Net Improve ServicePointManager.DefaultConnectionLimit "`ServicePointManager.DefaultConnectionLimit` appears to be `2` on .NET 4.5 in non-ASP.NET apps. This is too low for almost any production load. I don't know what this low limit is supposed to accomplish. I understand the presence of _some_ limit as a safety measure but `2` is really never useful. Many developers have trouble with that: https://www.google.com/webhp?complete=1&hl=en&gws_rd=cr,ssl&ei=#complete=1&hl=en&q=site:stackoverflow.com+%22ServicePointManager.DefaultConnectionLimit%22 When somebody asks ""My HTTP requests are unexpectedly slow"" the default answer is ""DefaultConnectionLimit = int.MaxValue"" and this solves the problem in about 80% of the cases. This is strong evidence that the default should be changed. This is an ongoing cause for trouble and surprise. There are three choices that I can think of: 1. Unlimited. ASP.NET has chosen this (`SetAutoConfigLimits`) presumably for good reasons. I think this is the least surprising choice. When a developer says `new WebClient().Download(...)` he expects the request to be sent right away. The dev is responsible for ensuring a sane degree of concurrency. he can use `SemaphoreSlim` or the service point infrastructure for that. 2. A constant factor on the CPU count of the machine (e.g. 10). Pro: This scales up with the size of the machine. Con: It might be a low-CPU (e.g. 2) machine that does little processing but relays it's request to some backend REST service that does the heavy lifting. 100 outstanding requests per CPU would not surprise me at all in such a situation. The benefits of async IO would be throttled. 3. `1`. At least this would make developers very quickly aware that the limit exists and that they need to change it. Maybe the value of `2` follows from that reasoning. I consider this to be a weak choice that is going to be perceived as stupid (as is `2`). Another reason for ""Unlimited"" is that there should not be global limits but per-service limits. A .NET process often hosts many components each of which potentially take from the global limit. They could interact in surprising (and spooky) ways. I think the .NET Framework should pick the default value that has the least surprise and set this to `int.MaxValue`. Also, consider revising the default value of `Expect100Continue`. I believe there is not a single developer in the world who expects each `POST` request to be performed twice by default. I do not know a single case where this was ever deemed useful. Disabling this behavior would entail, I believe, low compatibility risk. (See http://stackoverflow.com/search?q=Expect100Continue) "
```

File: docs/architecture/modernize-with-azure-containers/migrate-your-relational-databases-to-azure.md
Comment (ShannonLeavitt) [inline docs/architecture/modernize-with-azure-containers/migrate-your-relational-databases-to-azure.md:105]:
```suggestion
```
deleting blank line

File: docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv
Comment (ShannonLeavitt) [inline docs/machine-learning/tutorials/snippets/github-issue-classification/csharp/Data/issues_train.tsv:600]:
```suggestion
2329	area-System.Threading	Use a more dependable policy for thread pool thread injection	As of .NET 4.5 the thread pool injects one thread per 0.5sec if it thinks more threads are required. This is a problem if the number of required threads suddenly increases. Example: A corporate ASP.NET website is idle at night. In the morning at 8AM 1000 people log on and start working. If the app is using let's say 100 threads starting from 8AM it will take like 50sec to create all of them. Until then there will be serious delays and timeouts. It is possible to construe arbitrarily bad scenarios.  **Problem statement:** If thread pool load suddenly increases in IO bound workloads the pool is too slow to respond. This causes throughput and availability disruption. IO bound workloads relying on synchronous IO are common. Sudden workload changes are also common. Sometimes the workload can change due to a problem outside of the developer's control: A web service timing out or a database become slow.  Let me stress that this causes service interruption.  You can easily repro this yourself. Run a load test on an ASP.NET site with `Thread.Sleep(10000);`. The thread count goes up by 2 each second.  Starting and shutting down a thread was benchmarked by me to be around 1ms in total. Threads are not really an expensive resource. The thread pool should be a lot more eager to create and destroy threads. 500ms delay to potentially save 1ms is not a good trade-off.  **Easy fix:** I propose lowering the injection delay to 100ms. This reduces the problem given above by 5x. Ideally, the rate would be configurable. The shutdown delay could be lowered from 30s as well. Keeping an idle thread for 30000ms to save 1ms seems excessive. In this ticket I'm not that concerned with retiring threads, though.  **Smarter, riskier fix:** The delay could depend on the number of threads in existence, the core count and the perceived pressure on the thread-pool. The injection rate could be: - 0ms delay for up to (ProcessorCount \* 1) threads - 50ms delay for up to (ProcessorCount \* 4) threads - Starting from that a delay of (100ms \* (ThreadCount / ProcessorCount \* someFloatFactor)). Reasoning: The more the CPU is oversubscribed the slower we want to inject. Maybe we need to have a maximum delay of 1sec. Or, the delay must rise sub-linearly (e.g. sqrt).  Note, that the Windows Kernel has some thread pool injection heuristics that apply back pressure the more threads are in existence. This seems to work.	area-System.Net	Improve ServicePointManager.DefaultConnectionLimit	"`ServicePointManager.DefaultConnectionLimit` appears to be `2` on .NET 4.5 in non-ASP.NET apps. This is too low for almost any production load. I don't know what this low limit is supposed to accomplish. I understand the presence of _some_ limit as a safety measure but `2` is really never useful.  Many developers have trouble with that: https://www.google.com/webhp?complete=1&hl=en&gws_rd=cr,ssl&ei=#complete=1&hl=en&q=site:stackoverflow.com+%22ServicePointManager.DefaultConnectionLimit%22 When somebody asks ""My HTTP requests are unexpectedly slow"" the default answer is ""DefaultConnectionLimit = int.MaxValue"" and this solves the problem in about 80% of the cases. This is strong evidence that the default should be changed. This is an ongoing cause for trouble and surprise.  There are three choices that I can think of: 1. Unlimited. ASP.NET has chosen this (`SetAutoConfigLimits`) presumably for good reasons. I think this is the least surprising choice. When a developer says `new WebClient().Download(...)` he expects the request to be sent right away. The dev is responsible for ensuring a sane degree of concurrency. he can use `SemaphoreSlim` or the service point infrastructure for that. 2. A constant factor on the CPU count of the machine (e.g. 10). Pro: This scales up with the size of the machine. Con: It might be a low-CPU (e.g. 2) machine that does little processing but relays it's request to some backend REST service that does the heavy lifting. 100 outstanding requests per CPU would not surprise me at all in such a situation. The benefits of async IO would be throttled. 3. `1`. At least this would make developers very quickly aware that the limit exists and that they need to change it. Maybe the value of `2` follows from that reasoning. I consider this to be a weak choice that is going to be perceived as stupid (as is `2`).  Another reason for ""Unlimited"" is that there should not be global limits but per-service limits. A .NET process often hosts many components each of which potentially take from the global limit. They could interact in surprising (and spooky) ways.  I think the .NET Framework should pick the default value that has the least surprise and set this to `int.MaxValue`.  Also, consider revising the default value of `Expect100Continue`. I believe there is not a single developer in the world who expects each `POST` request to be performed twice by default. I do not know a single case where this was ever deemed useful. Disabling this behavior would entail, I believe, low compatibility risk. (See http://stackoverflow.com/search?q=Expect100Continue) "
```
Reverting the change

**Comment by ShannonLeavitt** [comment]
I have moved the commits in this PR to https://github.com/dotnet/docs/pull/29443

**Pull request closed without merging**
